### PR TITLE
fix: Allow borrowing if it brings TCR back up to CCR

### LINKED ISSUE #511: Recon M-04: Borrowing from branches can be disabled by one whale or early depositor
Below CCR, we disallow some borrower ops. A whale could bring TCR ~=CCR and DoS them.

### PR DESCRIPTION
**Pull request by bingen**
Fixes #511

### CODE REVIEW

**Comment by danielattilasimon** [comment]
So all along, opening a Trove below CCR was okay as long as it would tip the TCR above CCR? But borrowing against an existing Trove was a no-go. Well, it's good that we're making it consistent then.

**Comment by bingen** [comment]
No, I think it was already consistent, see here:
https://github.com/liquity/bold/pull/527/files#diff-d30cc87ed051c6909d6e9b48a0d719549c2272b1418110c6f55618d5ec6d9448L322

The aim is to fix the issue brought up by Alex, where a whale may block borrowing in general (both new and existing troves) by being able to place the system slightly below CCR.

**Comment by danielattilasimon** [comment]
Ah, I see, I missed the check being removed from `openTrove()`. Then we'll also need to change the `openTrove()` handler in the invariant test.

File: contracts/src/BorrowerOperations.sol [hunk 1]
```diff
@@ -1350,26 +1347,14 @@ contract BorrowerOperations is LiquityBase, AddRemoveManagers, IBorrowerOperatio
         }
     }
 
-    function _requireNotBelowCriticalThreshold(uint256 _price) internal view {
-        if (_checkBelowCriticalThreshold(_price, CCR)) {
-            revert BelowCriticalThreshold();
```
Comment (danielattilasimon) [inline contracts/src/BorrowerOperations.sol:1355]: This makes `error BelowCriticalThreshold()` unused, I believe.

Comment (bingen) [reply contracts/src/BorrowerOperations.sol:1355]: Oh, right, I removed `BorrowingNotPermittedBelowCT` but forgot this one. Thanks for fixing it!

**Comment by bingen** [comment]
And thanks for fixing the invariant tests too!

**Review by danielattilasimon — APPROVED** [review]
Looks good now!

**Pull request merged**